pygmt/helpers/testing.py

```diff
@@ -139,3 +140,41 @@ def wrapper(*args, ext="png", request=None, **kwargs):
         return wrapper
 
     return decorator
+
+
+def download_test_data():
+    """
+    Convenient function to download data files used in tests.
+    """
+    # List of earth_relief suffix to download
+    earth_relief_suffix = [
+        "01d_p",
+        "01d_g",
+        "30m_p",
+        "30m_g",
+        "10m_p",
+        "10m_g",
+        "05m_p",
+        "05m_g",
+    ]
+    # List of tiles of 03s/01s srtm data.
+    # Names like @N35E135.earth_relief_03s_g.nc is for internal use only.
+    # The naming scheme may change. DO NOT USE IT IN YOUR SCRIPTS.
+    earth_relief_tiles = ["@N35E135.earth_relief_03s_g.nc"]
+    # List of cache files
+    cache_files = [
+        "@fractures_06.txt",
+        "@ridge.txt",
+        "@srtm_tiles.nc",  # needed for 03s and 01s relief data
+        "@Table_5_11.txt",
+        "@test.dat.nc",
+        "@tut_bathy.nc",
+        "@tut_quakes.ngdc",
+        "@tut_ship.xyz",
+        "@usgs_quakes_22.txt",
+    ]
+
+    dataset = [f"@earth_relief_{suffix}" for suffix in earth_relief_suffix]
+    dataset.extend(earth_relief_tiles + cache_files)
+    for data in dataset:
+        which(data, download="a")
```

Comment:
The gmt `which` function can actually download multiple files at once (e.g. ` gmt which -Ga @ridge.txt @Table_5_11.txt @test.dat.nc`). Maybe better if we enable this in `pygmt.which` first instead of using a for-loop here that might be slower?

Reply:
Sounds a good idea.

Reply:
Ok, I'll open up a quick PR for it.

Reply:
#1312 is merged, so we can use `pygmt.which` to download a list of dataset(s). E.g. using

```suggestion
    which(fname=dataset, download="a")
```